src/FLACTagReader.js

```diff
@@ -0,0 +1,144 @@
+   * metadata, we must locate the header first. This can be done by reading the first byte of
+   * each block to determine the block type. After the block type comes a 24 bit integer that stores
+   * the length of the block as big endian. Using this, we locate the block and store the offset for
+   * parsing later.
+   *
+   * More info on the FLAC specification may be found here:
+   * https://xiph.org/flac/format.html
+   * @param {MediaFileReader} mediaFileReader - The MediaFileReader used to parse the file.
+   * @param {number} offset - The offset to start checking the header from.
+   * @param {LoadCallbackType} callbacks - The callback to call once the header has been found.
+   */
+  _loadBlock(
+    mediaFileReader: MediaFileReader,
+    offset: number,
+    callbacks: LoadCallbackType
+  ) {
+    var self = this;
+    var blockHeader = mediaFileReader.getByteAt(offset);
+    var blockSize = mediaFileReader.getInteger24At(offset + 1, true);
+    if (blockHeader !== 4 && blockHeader !== 132) {
+      mediaFileReader.loadRange([offset + 4 + blockSize, offset + 3 + 4 + blockSize], {
+        onSuccess: function() {
+          self._loadBlock(mediaFileReader, offset + 4 + blockSize, callbacks);
+        }
+      });
+    } else {
+      var offsetMetadata = offset + 4;
+      mediaFileReader.loadRange([offsetMetadata, offsetMetadata + blockSize], {
+        onSuccess: function() {
+          self._offset = offsetMetadata;
+          callbacks.onSuccess();
+        }
+      });
+    }
+  }
+
+  /**
+   * Parses the data and returns the tags.
+   * @param {MediaFileReader} data - The MediaFileReader to parse the file with.
+   * @param {Array<string>} [tags] - Optional tags to also be retrieved from the file.
+   * @return {TagType} - An object containing the tag information for the file.
+   */
+  _parseData(data: MediaFileReader, tags: ?Array<string>): TagType {
+    var string = data.getLongAt(this._offset, false);
+    var offsetVendor = this._offset + 4;
+    var vendor = data.getStringWithCharsetAt(offsetVendor, string, "utf-8").toString();
+    var offsetList = offsetVendor + string;
+    var length = data.getLongAt(offsetList, false);
+    var dataOffset = offsetList + 4;
+    var title, artist, album, track, genre, picture;
+    for (let i = 0; i < length; i++) {
+      let dataLength = data.getLongAt(dataOffset, false);
+      let s = data.getStringWithCharsetAt(dataOffset + 4, dataLength, "utf-8").toString();
+      let d = s.indexOf("=");
+      let split = [s.slice(0, d), s.slice(d + 1)];
+      switch (split[0]) {
+        case "TITLE":
+          title = split[1]
+        case "ARTIST":
+          artist = split[1]
+        case "ALBUM":
+          album = split[1]
+        case "TRACKNUMBER":
+          track = split[1]
+        case "GENRE":
+          genre = split[1]
+      }
+      dataOffset += 4 + dataLength;
+    }
+
+
+    var tag = {
+      type: "VorbisComment",
```

Comment:
I see, in this case I think it's better to use `FLAC` to follow the same convention as the other tag readers.